core/src/main/java/org/springframework/security/authorization/method/AuthorizationAnnotationPropertyConversionService.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2002-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.authorization.method;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.springframework.core.convert.ConversionService;
+import org.springframework.core.convert.TypeDescriptor;
+import org.springframework.core.convert.converter.ConverterRegistry;
+import org.springframework.core.convert.converter.GenericConverter;
+import org.springframework.core.convert.support.DefaultConversionService;
+import org.springframework.core.convert.support.GenericConversionService;
+
+/**
+ * A {@link ConversionService} configured with converters that provide class to string
+ * conversion
+ *
+ * @author DingHao
+ * @since 6.3
+ */
+public final class AuthorizationAnnotationPropertyConversionService extends GenericConversionService {
```

Comment:
While I appreciate the value of creating a custom implementation as `oauth2-core` does, let's start a little simpler for now, especially since this is on a maintenance branch.

What if instead, inside of `AuthorizationAnnotationUtils` you added something similar to:

```java
private final static DefaultConversionService conversionService = new DefaultConversionService();
static {
    conversionService.addConverter(...);
}
```

And then the code would use the `conversionService` instance.

Reply:
Done